testkit/src/lib.rs

```diff
@@ -465,10 +458,13 @@ impl TestKit {
     /// ```
     pub fn rollback(&mut self, blocks: usize) {
         assert!(
-            (blocks as u64) <= self.height().0,
+            self.height().0 >= blocks as u64,
             "Cannot rollback past genesis block"
         );
-        self.db_handler.rollback(blocks);
+        // Each block contain at least two phases:
```

Comment:
"At least two" isn't good enough, unfortunately. I think the best way to deal with the problem would be to add a method to the revertible database handler to manually report when new blocks are created (correspondingly, when new checkpoints should be created).